content/docs/learning-guide/tutorials/custom-tools/shape-example-cpp.md

```diff
@@ -0,0 +1,512 @@
+
+In this step, create an input field for the entity's name and a check box for an option to append a suffix---the component's name---to the entity's name. For example, suppose you set the entity's name to "MyEntity" and enable the check box. Then, when you create an entity with a **Box Shape** component and another with a **Sphere Shape** component, they will respectively be named "MyEntity_BoxShape" and "MyEntity_SphereShape".
+
+By the end of this step, your input field and check box should look like this: 
+![Shows UI for an input field and check box](/images/learning-guide/tutorials/custom-tools/shape-example/input-field-check-box.png)
+
+First, wrap these UI elements in their own sub-widget, set the layout, and add it to the main widget. 
+
+1. To start, instantiate a `QGroupBox` called `entityNameWidget` and a `QFormLayout` called `formLayout`.
+   
+2. Later, you will create the input field and check box and add them to this widget. 
+
+3. Finally, set the layout of `entityNameWidget` to `formLayout`, and add `entityNameWidget` to `mainLayout`. 
+
+```cpp
+    QWidget* entityNameWidget = new QWidget(this);  // 1
+    QFormLayout* formLayout = new QFormLayout();
+
+    // ...                                          // 2
+
+    entityNameWidget->setLayout(formLayout);        // 3
+    mainLayout->addWidget(entityNameWidget);
+```
+
+### Create an input field
+
+An input field is a UI element that takes text input from the user. With Qt, you can create an input field by using the `QLineEdit` object.
+
+1. Create an input field by instantiating `QLineEdit`. In this example, we'll name it `m_nameInput`.
+
+2. Set the placeholder text in `m_nameInput` by calling `setPlaceholderText(...)`.
+
+3. Enable a button to clear the text using `setClearButtonEnabled(true)`.
+
+
+```cpp
+    m_nameInput = new QLineEdit(this);
+    m_nameInput->setPlaceholderText(QObject::tr("Set custom Entity name here..."));
+    m_nameInput->setClearButtonEnabled(true);
+```
+
+### Create a check box
+A check box is an option button that users can enable or disable to trigger a user-defined behavior. With Qt, you can create an input field by using the `QCheckBox` object. In this example, the check box will control whether or not to append a suffix to the entity's name. At runtime, it will start disabled, and enable when the user enters the entity's name to the input field. You will define this behavior later.
+
+1. Create a check box by instantiating `QCheckBox`. In this example, name it `m_addShapeNameSuffix`.
+
+2. Set the check box to the start in the disabled state by using `setDisabled(true)`.
+
+```cpp
+    m_addShapeNameSuffix = new QCheckBox(this);
+    m_addShapeNameSuffix->setDisabled(true);
+```
+
+### Add a signal listener with a slot handler
+
+Qt uses signals and slots to communicate between objects. Set up a signal listener such that when the user enters text into the input field at runtime, the check box automatically enables. 
```

Comment:
Might want to have the text `signals and slots` be a link to the Qt docs on signals/slots:  https://doc.qt.io/qt-5/signalsandslots.html